performance_manager/lib/l1_full_trip_events.py

```diff
@@ -16,208 +13,230 @@
-            event_type.select_query
+    merged_query = (
+        sa.select(
+            move_vp_cte.c.fk_vp_moving_event,
+            stop_tu_cte.c.fk_tu_stopped_event,
+            stop_vp_cte.c.fk_vp_stopped_event,
+            sa.func.coalesce(
+                move_vp_cte.c.stop_sequence,
+                stop_vp_cte.c.stop_sequence,
+            ).label("stop_sequence"),
+            sa.func.coalesce(
+                move_vp_cte.c.stop_id, stop_vp_cte.c.stop_id
+            ).label("stop_id"),
+            sa.func.coalesce(
+                move_vp_cte.c.direction_id,
+                stop_vp_cte.c.direction_id,
+            ).label("direction_id"),
+            sa.func.coalesce(
+                move_vp_cte.c.route_id, stop_vp_cte.c.route_id
+            ).label("route_id"),
+            sa.func.coalesce(
+                move_vp_cte.c.start_date, stop_vp_cte.c.start_date
+            ).label("start_date"),
+            sa.func.coalesce(
+                move_vp_cte.c.start_time, stop_vp_cte.c.start_time
+            ).label("start_time"),
+            sa.func.coalesce(
+                move_vp_cte.c.vehicle_id, stop_vp_cte.c.vehicle_id
+            ).label("vehicle_id"),
         )
-
-        # if L0 select produces no results, create empty frame with expected
-        # columns "pk_id" & "hash"
-        if event_type.merge_dataframe.shape[0] == 0:
-            event_type.merge_dataframe = pandas.DataFrame(
-                columns=["pk_id", "hash"]
-            )
-        else:
-            # if data is retrieved from L0 table, ensure number columns are
-            # Int64 type to ensure consistent hashing
-            for column in convert_to_number_columns:
-                event_type.merge_dataframe[column] = pandas.to_numeric(
-                    event_type.merge_dataframe[column]
-                ).astype("Int64")
-            # create hash column to be used as FullTripEvent primary key
-            # drop category columns after hash has been created
-            event_type.merge_dataframe = add_event_hash_column(
-                event_type.merge_dataframe,
-                expected_hash_columns=expected_hash_columns,
-            )[["pk_id", "hash"]]
-
-        # rename pk_id field from L0 table to L1 foreign key field name
-        event_type.merge_dataframe.rename(
-            columns={"pk_id": event_type.fk_rename_field}, inplace=True
+        .join(
+            stop_tu_cte,
+            sa.and_(
+                move_vp_cte.c.stop_sequence == stop_tu_cte.c.stop_sequence,
+                move_vp_cte.c.stop_id == stop_tu_cte.c.stop_id,
+                move_vp_cte.c.direction_id == stop_tu_cte.c.direction_id,
+                move_vp_cte.c.route_id == stop_tu_cte.c.route_id,
+                move_vp_cte.c.start_date == stop_tu_cte.c.start_date,
+                move_vp_cte.c.start_time == stop_tu_cte.c.start_time,
+                move_vp_cte.c.vehicle_id == stop_tu_cte.c.vehicle_id,
```

Comment:
is this section doing the work that the hash was doing before?

ensuring that all this data is the same?